beginend.el

```diff
@@ -232,6 +232,32 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+;; Copy/pasted from expand-region
+(defun beginend--point-is-in-comment-p ()
+  "Return non-nil if point is in comment."
+  (or (nth 4 (syntax-ppss))
+      (memq (get-text-property (point) 'face)
+            '(font-lock-comment-face font-lock-comment-delimiter-face))))
```

Comment:
This is very problematic.  I used to have similar implementation in smartparens but as the buffers grow largrer and jit-fontlock kicks in things break.  And if you force the entire buffer to be fontified prior to the check it will get slooooooow.

Smartparens now uses this, which is rather involved:

```elisp
(defun sp-point-in-comment (&optional p)
  "Return non-nil if point is inside comment.

If optional argument P is present test this instead off point."
  (setq p (or p (point)))
  (ignore-errors
    (save-excursion
      (or (nth 4 (sp--syntax-ppss p))
          ;; this also test opening and closing comment delimiters... we
          ;; need to chack that it is not newline, which is in "comment
          ;; ender" class in elisp-mode, but we just want it to be
          ;; treated as whitespace
          (and (< p (point-max))
               (memq (char-syntax (char-after p)) '(?< ?>))
               (not (eq (char-after p) ?\n)))
          ;; we also need to test the special syntax flag for comment
          ;; starters and enders, because `syntax-ppss' does not yet
          ;; know if we are inside a comment or not (e.g. / can be a
          ;; division or comment starter...).
          (-when-let (s (car (syntax-after p)))
            (or (and (/= 0 (logand (lsh 1 16) s))
                     (nth 4 (syntax-ppss (+ p 2))))
                (and (/= 0 (logand (lsh 1 17) s))
                     (nth 4 (syntax-ppss (+ p 1))))
                (and (/= 0 (logand (lsh 1 18) s))
                     (nth 4 (syntax-ppss (- p 1))))
                (and (/= 0 (logand (lsh 1 19) s))
                     (nth 4 (syntax-ppss (- p 2))))))))))
```

I was thinking of packaging this up because I have found the need to use it in a couple other projects as well.